dataconfig/parsers.py

```diff
@@ -68,7 +69,32 @@ def from_yaml(cls, yaml_path: Union[str, Path]) -> _ConfigIO:
         # FIXME: type checking is ignored for the return statement because mypy
         # doesn't seem to know this is an abstract base class, and the argument
         # unpacking makes sense when instantiating any of the derived classes.
-        return cls(**read_yaml(yaml_path))  # type: ignore
+
+        # read in the config
+        conf = read_yaml(yaml_path)
+        
+        # get the leafs of the config file -> these are the values we potentially need to apply the transformations to
+        # leaves is an Iterable[_Path_t]
+        leaves = _leaves(_nodes(conf))
```

Comment:
This looks okay, but maybe you need to filter with `hasattr(obj, "transform")`.  Then possibly the block below would be unneeded.